### PR TITLE
Add API for Retrieving All Documents

### PR DESCRIPTION
**Pull request by raararaara**
**What this PR does / why we need it**:
This PR introduces additional functionality required for the creation and retrieval of documents through REST API. Specifically, it adds the following enhancements:

* Authentication via Project Secret Key: Previously, authentication for API access was limited to user JWT tokens only. With this update, we have extended the authentication mechanism to also allow authentication using a Project Secret Key. This change provides more flexibility in how users and systems can authenticate and access our API.

**Which issue(s) this PR fixes**:

Related #677 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: server/backend/database/database.go [hunk 1]
```diff
@@ -118,6 +124,9 @@ type Database interface {
 	// FindUserInfo returns a user by the given username.
 	FindUserInfo(ctx context.Context, username string) (*UserInfo, error)
 
+	// FindUserInfoById finds a user by the given id.
```
Comment (hackerwins) [inline server/backend/database/database.go:127]:
```suggestion
	// FindUserInfoByID finds a user by the given id.
```

File: server/projects/projects.go [hunk 2]
```diff
@@ -107,3 +107,21 @@ func GetProjectFromAPIKey(ctx context.Context, be *backend.Backend, apiKey strin
...
+
+// GetProjectFromSecretKey returns a project from a secret key.
+func GetProjectFromSecretKey(ctx context.Context, be *backend.Backend, secretKey string) (*types.Project, error) {
+	if secretKey == "" {
+		info, err := be.DB.FindProjectInfoByID(ctx, database.DefaultProjectID)
+		if err != nil {
+			return nil, err
+		}
+		return info.ToProject(), nil
+	}
```
Comment (hackerwins) [inline server/projects/projects.go:119]: I think this logic returns the default project if we don't pass a secretKey. Why is this needed?

File: server/backend/database/database.go [hunk 3]
```diff
@@ -118,6 +124,9 @@ type Database interface {
...
+	FindUserInfoByID(ctx context.Context, id string) (*UserInfo, error)
```
Comment (hackerwins) [inline server/backend/database/database.go:128]: We use `types.ID` to identify the resource in DB. Is it possible to change this?

```suggestion
	FindUserInfoByID(ctx context.Context, id types.ID) (*UserInfo, error)
```

File: server/rpc/interceptors/admin_auth.go [hunk 4]
```diff
@@ -162,14 +163,20 @@ func (i *AdminAuthInterceptor) authenticate(
...
+		user, err := users.GetUser(ctx, i.backend, claims.Username)
+		if err == nil {
+			return user, nil
+		}
 	}
 
-	user, err := users.GetUser(ctx, i.backend, claims.Username)
-	if err != nil {
-		return nil, connect.NewError(connect.CodeUnauthenticated, ErrUnauthenticated)
+	project, err := projects.GetProjectFromSecretKey(ctx, i.backend, authorization)
```
Comment (hackerwins) [inline server/rpc/interceptors/admin_auth.go:173]:
```suggestion
	// NOTE(hackerwins): If the token is secret key, return the owner of the project.
	project, err := projects.GetProjectFromSecretKey(ctx, i.backend, authorization)
```

File: server/rpc/interceptors/admin_auth.go [hunk 5]
```diff
@@ -162,14 +163,20 @@ func (i *AdminAuthInterceptor) authenticate(
 	}
 
 	claims, err := i.tokenManager.Verify(authorization)
```
Comment (hackerwins) [inline server/rpc/interceptors/admin_auth.go:165]:
```suggestion
	// NOTE(hackerwins): If the token is access token, return the user of the token.
	claims, err := i.tokenManager.Verify(authorization)
```

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thanks for your contribution. Overall looks good.

I left a few comments.

File: server/projects/projects.go (hunk 2 above)
Comment (raararaara) [reply server/projects/projects.go:119]: I was also curious about the necessity of this logic, and since `GetProjectFromAPIKey` was returning the default project in the same way, I thought there must be a reason, so I did the same.
I don't particularly think of any side effects that occur when the default project is not returned.

File: server/backend/database/database.go (hunk 3 above)
Comment (raararaara) [reply server/backend/database/database.go]: Initially, it was implemented based on `types.ID`, but when testing in memoryDB, finding user by clientID with type `types.ID` did not work correctly. 

This was solved by converting `types.ID` to `string` when querying memoryDB.

File: server/projects/projects.go (hunk 2 above)
Comment (hackerwins) [reply server/projects/projects.go:119]: To get started with Yorkie, users can easily install it using a package manager like Homebrew and they can run Yokie in standalone mode. We allows users to test the functionality without the need to create a user or project explicitly to get the apiKey.

But I don't think this feature(return default project without apiKey) needs for secretKey.

Comment (hackerwins) [reply server/projects/projects.go]: https://github.com/yorkie-team/yorkie/pull/755/commits/b3a93275abf0d17c6ad65ff3f6f7bee7e5b93d7c

**Review by hackerwins — APPROVED** [review]
LGTM. 🚀
